magic_the_gathering/cards/deck_creator.py

```diff
@@ -32,18 +32,31 @@ def create_decks(self, n_players: int) -> List[OrderedDict[str, Card]]:
         for _ in range(n_players):
             current_deck = OrderedDict()
             color = random.choice(["W", "U", "B", "R", "G"])
-            land_color_filter = self.legal_lands_df["color_identity"] == f"['{color}']"
-            land_card_names = self.legal_lands_df[land_color_filter]["name"].unique()
-            creature_color_filter = self.legal_creatures_df["color_identity"] == f"['{color}']"
-            creature_card_names = self.legal_creatures_df[creature_color_filter]["name"].unique()
             for _ in range(self.deck_size):
-                if random.random() < self.lands_proportion:
-                    card_name = random.choice(land_card_names)
-                    card_series = self.legal_lands_df[self.legal_lands_df["name"] == card_name].iloc[0]
-                else:
-                    card_name = random.choice(creature_card_names)
-                    card_series = self.legal_creatures_df[self.legal_creatures_df["name"] == card_name].iloc[0]
-                card = Card.from_series(card_series)
-                current_deck[card.uuid] = card
+                self.add_a_card_at_random(color, current_deck)
             decks.append(current_deck)
         return decks
+
+    def add_a_card_at_random(
+        self,
+        color: str,
+        current_deck: OrderedDict[str, Card],
+        log: bool = False,
+    ) -> OrderedDict[str, Card]:
+        creature_color_filter = self.legal_creatures_df["color_identity"] == f"['{color}']"
+        creature_card_names = self.legal_creatures_df[creature_color_filter]["name"].unique()
+        land_color_filter = self.legal_lands_df["color_identity"] == f"['{color}']"
+        land_card_names = self.legal_lands_df[land_color_filter]["name"].unique()
```

Comment:
These variables are relatively constant (they only vary depending on the color) so it would probably be more optimal to create them once and reuse them in `add_a_card_at_random` (currently if I understand well, they are recomputed at each iteration of the for loop above